dev-tools/deploy/preflight.py

```
@@ -442,15 +567,62 @@ def validate_recovery_baseline(
-    ):
+    if not isinstance(backup_artifact_lineage, dict):
+        return (
+            "fail",
+            "Recovery compatibility baseline artifact lineage must be an object",
+        )
+    pre_snapshot_journal_high_water = backup_artifact_lineage.get("preSnapshotJournalHighWater")
+    if backup_artifact_lineage.get("artifactErasureHighWater") != artifact_high_water:
+        return (
+            "fail",
+            "Recovery compatibility baseline artifact lineage artifactErasureHighWater "
+            "must match the snapshot-bound artifact high-water sequence",
```

Ruff (0.16.0)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)